thcrap_proton

```diff
@@ -71,6 +75,40 @@ if [ ! -f "$THCRAP_FOLDER/config/$THCRAP_CONFIG" ]; then
     fi
 fi
 
+if [ "$USE_THPRAC" == 1 ]; then
+    THPRAC_VERSION=$(LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl --connect-timeout 5 -si https://github.com/touhouworldcup/thprac/releases/latest | grep location: | sed 's@.*/@@' | tr -d '[:space:]')
+    if [[ -n "$THPRAC_VERSION" && ! -f "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" ]]; then
+        # Install latest version
+        mkdir -p "$THPRAC_FOLDER"
+        if [ ! -w "$THPRAC_FOLDER" ]; then
+            ${ZENITY} --error --text="The directory \"$THPRAC_FOLDER\" could not be created or is not writable. Check your permissions and try again."
+            exit
+        fi
+        LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl -L -f "https://github.com/touhouworldcup/thprac/releases/download/$THPRAC_VERSION/thprac.$THPRAC_VERSION.exe" -o "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" | ${ZENITY} --progress --pulsate --auto-close --text="Downloading \"thprac.$THPRAC_VERSION.exe\""
+        if [ $? != 0 ]; then
+            # Only abort if there's no version currently installed
+            if [ ! -f "$THPRAC_FOLDER/thprac.exe" ]; then
+                ${ZENITY} --error --text="Download failed. Aborting"
+                exit
+            fi
+        else
+            # Remove old versions, only keep 5 latest versions
+            readarray -d '' THPRAC_OLD_VERSIONS < <(find "$THPRAC_FOLDER" -maxdepth 1 -regextype egrep -iregex '.*/thprac.v([0-9]|\.)*.exe' -print0 | sort -zVr | tail -z -n +6)
+            if (( ${#THPRAC_OLD_VERSIONS[@]} )); then
+                # THPRAC_OLD_VERSIONS not empty
+                rm "${THPRAC_OLD_VERSIONS[@]}"
+            fi
+            # Create a symlink to latest version
+            # You can manually change launch version by pointing the symlink to old version.
```

Comment:
Is there any reason to run an old version? (I genuinely ask since I never used the tool)

Reply:
Well, it's also my first time using thprac. I thought this could roll back, in case bug in latest release. Probably it's not a good approach. I could remove this comment if you prefer.

Reply:
Yeah I don't think it makes sense.